code/libraries/BigQuery/BigQuery.js

```diff
@@ -1,301 +1,346 @@
+         * For more Information regarding optional parameters and response structure:
+         * https://cloud.google.com/bigquery/docs/reference/rest/v2/datasets/delete             *
+         * 
+         * @param {callback} callback Provide a function with signature: function(err, response) 
+         * 
+         * @example 
+            dataset.deleteDataset(function(err, response){
+                if(err){
+                    resp.error(err);
+                }
+                resp.success(response);
+            });
+         * 
+         */
+        function deleteDataset(callback) {
+            var reqOptions = _createRequestObject(URL_WITH_CURRENT_DATASET);
+            http.delete(reqOptions, callback);
+        }
+       
+        /**
+        * @typedef {Object} DatasetResource
+        * {@link https://cloud.google.com/bigquery/docs/reference/rest/v2/datasets#resource}
+        */
 
-/**
- * Tabledata: insertAll
- * insertAll - Streams data into BigQuery one record at a time without needing to run a load job. 
- * 
- * For more information: https://cloud.google.com/bigquery/docs/reference/rest/v2/tabledata/insertAll
- * 
- * @param {string} projectID
- * @param {string} datasetID 
- * @param {string} tableID - tableID in the dataset selected above
- * @param {Object} requestBody - 
- * {
- *   "rows":
- *  [
- *    {
- *      "json":
- *      v{
- *        "column1": "value",
- *        "column2": "value"
- *      }
- *    }
- *  ]
- * }
- * @param {callback} callback - callback with function signature: (err, data)
- * 
- */
-BigQuery.prototype.insertAll = function(projectID, datasetID, tableID, requestBody, callback){
-    if(this.key === ""){
-        callback(true, "Must provide API Key before making requests to Google BigQuery Service.");
-    }
-    var authToken = "Bearer "+ this.key;
-    
-    var options = {
-    	uri: "https://www.googleapis.com/bigquery/v2/projects/" + projectID + "/datasets/" + datasetID + "/tables/" + tableID + "/insertAll",
-    	headers: {
-    	    "Content-Type": "application/json",
-    		"Authorization": authToken
-    	},
-    	body: requestBody
-    }    
-    
-    requestObject.post(options, function(err, data) {
-        if(!err){
-            callback(false, data);
+        /**
+        * Updates information in an existing dataset. The update method replaces the entire dataset resource,
+        * whereas the patch method only replaces fields that are provided in the submitted dataset resource.
+        * For more Information regarding optional parameters and response structure: 
+        * https://cloud.google.com/bigquery/docs/reference/rest/v2/datasets/update
+        *
+        * @param {DatasetResource} requestBody - The DatasetResource should contain the reference dataset 
+        * @param {callback} callback Provide a function with signature: function(err, response) 
+        * 
+        * @example
+            var requestBody = {
```

Comment:
This example is very confusing, and not valid JavaScript. provide a valid JSON example